src/test/java/org/broadinstitute/hellbender/utils/MathUtilsUnitTest.java

```diff
@@ -605,4 +605,380 @@ public void testMedianOfEmptyList(){
+                new int[]{4,2,1,3},
+                new int[]{4,2,3,1},
+                new int[]{4,3,1,2},
+                new int[]{4,3,2,1} };
+        for ( int gen = 0; gen < gens.length; gen ++ ) {
+            for ( int idx = 0; idx < 3; idx++ ) {
+                Assert.assertEquals(arr[idx],gens[gen][idx],
+                        String.format("Error at generation %d, expected %s, observed %s",gen,Arrays.toString(gens[gen]),Arrays.toString(arr)));
+            }
+            arr = nextPermutation(arr);
+        }
+    }
+
+    private double[] addEpsilon(double[] counts) {
+        double[] d = new double[counts.length];
+        for ( int i = 0; i < counts.length; i ++ ) {
+            d[i] = counts[i] + 1e-3;
+        }
+        return d;
+    }
+
+    @Test
+    public void testDirichletMultinomial() {
+        List<double[]> testAlleles = Arrays.asList(
+                new double[]{80,240},
+                new double[]{1,10000},
+                new double[]{0,500},
+                new double[]{5140,20480},
+                new double[]{5000,800,200},
+                new double[]{6,3,1000},
+                new double[]{100,400,300,800},
+                new double[]{8000,100,20,80,2},
+                new double[]{90,20000,400,20,4,1280,720,1}
+        );
+
+        Assert.assertTrue(! Double.isInfinite(MathUtils.log10Gamma(1e-3)) && ! Double.isNaN(MathUtils.log10Gamma(1e-3)));
+
+        int[] numAlleleSampled = new int[]{2,5,10,20,25};
+        for ( double[] alleles : testAlleles ) {
+            for ( int count : numAlleleSampled ) {
+                // test that everything sums to one. Generate all multinomial draws
+                List<Double> likelihoods = new ArrayList<>(100000);
+                NextCounts generator = new NextCounts(alleles.length,count);
+                double maxLog = Double.MIN_VALUE;
+                //List<String> countLog = new ArrayList<String>(200);
+                while ( generator.hasNext() ) {
+                    int[] thisCount = generator.next();
+                    //countLog.add(Arrays.toString(thisCount));
+                    Double likelihood = MathUtils.dirichletMultinomial(addEpsilon(alleles),thisCount);
+                    Assert.assertTrue(! Double.isNaN(likelihood) && ! Double.isInfinite(likelihood),
+                            String.format("Likelihood for counts %s and nAlleles %d was %s",
+                                    Arrays.toString(thisCount),alleles.length,Double.toString(likelihood)));
+                    if ( likelihood > maxLog )
+                        maxLog = likelihood;
+                    likelihoods.add(likelihood);
+                }
+                //System.out.printf("%d likelihoods and max is (probability) %e\n",likelihoods.size(),Math.pow(10,maxLog));
+                Assert.assertEquals(MathUtils.sumLog10(unwrap(likelihoods)),1.0,1e-7,
+                        String.format("Counts %d and alleles %d have nLikelihoods %d. \n Counts: %s",
```

Comment:
\[FindBugs\] INFO: FS: Format string should use %n rather than \n in org\.broadinstitute\.hellbender\.utils\.MathUtilsUnitTest\.testDirichletMultinomial\(\)